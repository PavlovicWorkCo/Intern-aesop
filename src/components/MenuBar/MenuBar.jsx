import React from 'react';
import PropTypes from 'prop-types';
import FakeLink from '../FakeLink/FakeLink';
import './MenuBar.scss';
import arrowPicture from '../../assets/left-arrow.svg';

class MenuBar extends React.PureComponent {
  constructor() {
    super();
    this.state = {
      activeMenuBarOption: null,
    };
  }

  setActiveMenuBarOption(value) {
    this.setState({
      activeMenuBarOption: value,
    });
  }

  handleFakeLinkMouseEnter(menuBarOption) {
    const { menuIsActive, showMenuList, windowSize } = this.props;
    if (windowSize === 'small') return;
    if (menuIsActive) {
      this.setActiveMenuBarOption(menuBarOption);
    }
    showMenuList(menuBarOption);
  }

  handleFakeLinkClick(menuBarOption) {
    const { openMenu, showMenuList } = this.props;
    this.setActiveMenuBarOption(menuBarOption);
    showMenuList(menuBarOption);
    openMenu();
  }

  handleFakeLinkMouseLeave() {
    const { menuIsActive, windowSize } = this.props;
    if (windowSize === 'small') return;
    if (menuIsActive) return;
    this.setActiveMenuBarOption(null);
  }

  renderMenuBarOptions() {
    const { activeMenuBarOption } = this.state;
    const { handleCloseMenuClick, menuIsActive } = this.props;
    const menuBarOptionsArray = ['Shop', 'Read'];
    const closeButtonVisibleClass = menuIsActive ? 'Visible' : '';
    const menuBarOptions = menuBarOptionsArray.map((menuBarOption) => {
      const activeClass = menuBarOption === activeMenuBarOption && menuIsActive ? 'Active' : '';
      return (
        <FakeLink
          text={menuBarOption}
          className={`Menu-link ${activeClass}`}
          onClick={() => this.handleFakeLinkClick(menuBarOption)}
          onMouseEnter={() => this.handleFakeLinkMouseEnter(menuBarOption)}
          onMouseLeave={() => this.handleFakeLinkMouseLeave()}
          key={menuBarOption}
        />
      );
    });
    return (
      <div className="Menu-bar-left">
        <div className="Menu-bar-options">
          {menuBarOptions}
        </div>
        <button type="button" tabIndex={menuIsActive ? 0 : -1} className={`Close-menu ${closeButtonVisibleClass}`} onClick={handleCloseMenuClick} />
      </div>
    );
  }

  render() {
    const { className, menuIsActive, windowSize, handleBackButton } = this.props;
    const toggleMenuActive = menuIsActive ? 'Active' : '';
    return (
      <div className={className}>
        {windowSize === 'small' && (
          <button
            onClick={() => handleBackButton()}
            type="button"
            className={`Menu-back-button ${toggleMenuActive}`}
          ><img src={arrowPicture} /> </button>
        )}
        {windowSize !== 'small' && this.renderMenuBarOptions()}
        <div>
          <FakeLink tabIndex={menuIsActive ? -1 : 0} text="Login" className="Menu-link Login-button" />
          <button
            className={`Toggle-menu ${toggleMenuActive}`}
            type="button" onClick={() => this.handleFakeLinkClick('Toggle-menu')}
          />
        </div>
      </div>
    );
  }
}


MenuBar.defaultProps = {
  menuIsActive: false,
  openMenu: null,
  showMenuList: null,
  className: 'Menu-bar',
  handleCloseMenuClick: null,
  windowSize: null,
  handleBackButton: null,
};

MenuBar.propTypes = {
  menuIsActive: PropTypes.bool,
  openMenu: PropTypes.func,
  showMenuList: PropTypes.func,
  className: PropTypes.string,
  handleCloseMenuClick: PropTypes.func,
  windowSize: PropTypes.string,
  handleBackButton: PropTypes.func,
};

export default MenuBar;
