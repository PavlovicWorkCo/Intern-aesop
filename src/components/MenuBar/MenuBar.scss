.Menu-bar {
  left: 30px;
  top: 0px;
  display: inline-block;
  position: absolute;
  z-index: 6;
  margin-top: 20px;

  .Menu-link {
    color: black;
    display: inline-block;
    margin-right: 30px;
    &.Active::after {
      width: 100%;
    }
    &::after {
      content: '';
      width: 0px;
      height: 1px;
      display: block;
      background-color: black;
      transition: width 0.5s;
    }
    &:hover::after {
      width: 100%;
    }
  }
}
