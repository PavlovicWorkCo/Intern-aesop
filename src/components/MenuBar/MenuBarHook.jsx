import React, { useState } from 'react';
import PropTypes from 'prop-types';
import FakeLink from '../FakeLink/FakeLink';
import arrowPicture from '../../assets/left-arrow.svg';
import './MenuBar.scss';

export default function MenuBarHook({
  windowSize, className, menuIsActive,
  handleBackButton, showMenuList, openMenu, handleCloseMenuClick,
}) {
  const [activeMenuBarOption, setActiveMenuBarOption] = useState(null);

  function handleFakeLinkMouseEnter(menuBarOption) {
    if (windowSize === 'small') return;
    if (menuIsActive) {
      setActiveMenuBarOption(menuBarOption);
    }
    showMenuList(menuBarOption);
  }

  function handleFakeLinkClick(menuBarOption) {
    setActiveMenuBarOption(menuBarOption);
    showMenuList(menuBarOption);
    openMenu();
  }

  function handleFakeLinkMouseLeave() {
    if (windowSize === 'small') return;
    if (menuIsActive) return;
    setActiveMenuBarOption(null);
  }

  function renderMenuBarOptions() {
    const menuBarOptionsArray = ['Shop', 'Read'];
    const closeButtonVisibleClass = menuIsActive ? 'Visible' : '';
    const menuBarOptions = menuBarOptionsArray.map((menuBarOption) => {
      const activeClass = menuBarOption === activeMenuBarOption && menuIsActive ? 'Active' : '';
      return (
        <FakeLink
          text={menuBarOption}
          className={`Menu-link ${activeClass}`}
          onClick={() => handleFakeLinkClick(menuBarOption)}
          onMouseEnter={() => handleFakeLinkMouseEnter(menuBarOption)}
          onMouseLeave={() => handleFakeLinkMouseLeave()}
          key={menuBarOption}
        />
      );
    });
    return (
      <div className="Menu-bar-left">
        <div className="Menu-bar-options">
          {menuBarOptions}
        </div>
        <button type="button" tabIndex={menuIsActive ? 0 : -1} className={`Close-menu ${closeButtonVisibleClass}`} onClick={handleCloseMenuClick} />
      </div>
    );
  }

  const toggleMenuActive = menuIsActive ? 'Active' : '';
  return (
    <div className={className}>
      {windowSize === 'small' && (
        <button
          onClick={() => handleBackButton()}
          type="button"
          className={`Menu-back-button ${toggleMenuActive}`}
        >
          <img alt="left arrow" src={arrowPicture} />
        </button>
      )}
      {windowSize !== 'small' && renderMenuBarOptions()}
      <div>
        <FakeLink tabIndex={menuIsActive ? -1 : 0} text="Login" className="Menu-link Login-button" />
        <button
          className={`Toggle-menu ${toggleMenuActive}`}
          type="button"
          onClick={() => handleFakeLinkClick('Toggle-menu')}
        />
      </div>
    </div>
  );
}

MenuBarHook.defaultProps = {
  menuIsActive: false,
  openMenu: null,
  showMenuList: null,
  className: 'Menu-bar',
  handleCloseMenuClick: null,
  windowSize: null,
  handleBackButton: null,
};

MenuBarHook.propTypes = {
  menuIsActive: PropTypes.bool,
  openMenu: PropTypes.func,
  showMenuList: PropTypes.func,
  className: PropTypes.string,
  handleCloseMenuClick: PropTypes.func,
  windowSize: PropTypes.string,
  handleBackButton: PropTypes.func,
};
