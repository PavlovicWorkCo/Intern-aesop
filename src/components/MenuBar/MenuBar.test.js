import React from 'react';
import MenuBar from './MenuBar';
import { create, update } from 'react-test-renderer';

describe('My first snapshot test', () => {
    test('testing MenuBar snapshot', () => {
        let tree = create(<MenuBar />)
        expect(tree.toJSON()).toMatchSnapshot();
    })
})

describe('Change the activeMenuBarOption value', () => {
    test('testing MenuBar snapshot', () => {
        let tree = create(<MenuBar />)
        let instance = tree.getInstance();
        expect(instance.state.activeMenuBarOption).toBe(null);
        instance.setActiveMenuBarOption('Link text');
        expect(instance.state.activeMenuBarOption).toBe('Link text');
        expect(tree.toJSON()).toMatchSnapshot();

    })
})
