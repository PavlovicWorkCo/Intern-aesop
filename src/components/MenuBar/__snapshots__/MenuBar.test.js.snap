// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change the activeMenuBarOption value testing MenuBar snapshot 1`] = `
<div
  className="Menu-bar"
>
  <div
    className="Menu-bar-left"
  >
    <div
      className="Menu-bar-options"
    >
      <a
        className="Menu-link "
        href="#"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        tabIndex={null}
      >
        <span>
          Shop
        </span>
      </a>
      <a
        className="Menu-link "
        href="#"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        tabIndex={null}
      >
        <span>
          Read
        </span>
      </a>
    </div>
    <button
      className="Close-menu "
      onClick={null}
      tabIndex={-1}
      type="button"
    />
  </div>
  <div>
    <a
      className="Menu-link Login-button"
      href="#"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      tabIndex={0}
    >
      <span>
        Login
      </span>
    </a>
    <button
      className="Toggle-menu "
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`My first snapshot test testing MenuBar snapshot 1`] = `
<div
  className="Menu-bar"
>
  <div
    className="Menu-bar-left"
  >
    <div
      className="Menu-bar-options"
    >
      <a
        className="Menu-link "
        href="#"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        tabIndex={null}
      >
        <span>
          Shop
        </span>
      </a>
      <a
        className="Menu-link "
        href="#"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        tabIndex={null}
      >
        <span>
          Read
        </span>
      </a>
    </div>
    <button
      className="Close-menu "
      onClick={null}
      tabIndex={-1}
      type="button"
    />
  </div>
  <div>
    <a
      className="Menu-link Login-button"
      href="#"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      tabIndex={0}
    >
      <span>
        Login
      </span>
    </a>
    <button
      className="Toggle-menu "
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
